core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.directive.html

```diff
@@ -13,48 +15,169 @@ <h2 class="dashboard-title" translate="I18N_DASHBOARD_TOPICS_AND_SKILLS_DASHBOAR
+            <div class="dashboard-list-header-count">
+              <span>Displaying <[$ctrl.pageNumber*$ctrl.itemsPerPage + 1]> - <[$ctrl.pageNumber*$ctrl.itemsPerPage+10]> of <[$ctrl.currentCount]></span>
+            </div>
+            <div class="dashboard-list-pagination">
+              <div>
+                <select ng-options="choice as choice for choice in $ctrl.itemsPerPageChoice"
+                        ng-model="$ctrl.itemsPerPage"
+                        ng-change="$ctrl.refreshPagination()">
+                </select>
+              </div>
+              <div>
+                <i class="fa fa-angle-left topic-pagination-arrow" ng-click="$ctrl.changePageByOne($ctrl.MOVE_TO_PREV_PAGE)"></i>
+                <span class="dashboard-list-pagination-value" ng-repeat="n in $ctrl.generateNumbersTillRange($ctrl.currentCount/$ctrl.itemsPerPage)" ng-click="$ctrl.changePage($index)">
+                  <span ng-if="$ctrl.pageNumber===$index" class="dashboard-list-pagination-active"><[$index+1]></span>
+                  <span ng-if="$ctrl.pageNumber!==$index"><[$index+1]></span>
+                </span>
+                <i class="fa fa-angle-right topic-pagination-arrow" ng-click="$ctrl.changePageByOne($ctrl.MOVE_TO_NEXT_PAGE)"></i>
+              </div>
+            </div>
+          </div>
 
-    <div ng-if="$ctrl.activeTab === $ctrl.TAB_NAME_TOPICS">
-      <topics-list topic-summaries="$ctrl.topicSummaries"
-                   ng-if="$ctrl.topicSummaries.length > 0"
-                   user-can-delete-topic="$ctrl.userCanDeleteTopic">
-      </topics-list>
-      <p class="intro-card-message" ng-if="$ctrl.isTopicTabHelpTextVisible()" style="padding-top: 6px;">
-        No topics have been created yet.
-      </p>
-    </div>
-
-    <div ng-if="$ctrl.activeTab === $ctrl.TAB_NAME_UNTRIAGED_SKILLS">
-      <skills-list skill-summaries="$ctrl.untriagedSkillSummaries"
-                   ng-if="$ctrl.untriagedSkillSummaries.length > 0"
-                   editable-topic-summaries="$ctrl.editableTopicSummaries"
-                   mergeable-skill-summaries="$ctrl.mergeableSkillSummaries"
-                   user-can-create-skill="$ctrl.userCanCreateSkill"
-                   user-can-delete-skill="$ctrl.userCanDeleteSkill">
-      </skills-list>
-      <p class="intro-card-message" ng-if="$ctrl.isSkillsTabHelpTextVisible()" style="padding-top: 6px;">
-        There are no pending unused skills.
-      </p>
+          <div ng-if="$ctrl.activeTab === $ctrl.TAB_NAME_TOPICS">
+            <topics-list topic-summaries="$ctrl.displayedTopicSummaries"
+                         page-number="$ctrl.pageNumber"
+                         items-per-page="$ctrl.itemsPerPage"
+                         ng-if="$ctrl.topicSummaries.length > 0"
+                         user-can-delete-topic="$ctrl.userCanDeleteTopic">
+            </topics-list>
+            <p class="intro-card-message" ng-if="!$ctrl.topicSummaries.length" style="padding-top: 6px;">
```

Comment:
I see this style property being repeated. Can you put it in a class and use that instead?

Reply:
Done.